pkg/cmd/server/origin/auth.go

```diff
@@ -443,6 +444,9 @@ func (c *AuthConfig) getOAuthProvider(identityProvider configapi.IdentityProvide
 	case (*configapi.GitHubIdentityProvider):
 		return github.NewProvider(identityProvider.Name, provider.ClientID, provider.ClientSecret), nil
 
+	case (*configapi.GitLabIdentityProvider):
+		return gitlab.NewProvider(identityProvider.Name, provider.HostURL, provider.ClientID, provider.ClientSecret)
```

Comment:
follow the example of OpenIDIdentityProvider in building a transport from `provider.CA`